homeassistant/helpers/intent.py

```diff
@@ -117,6 +124,60 @@ def __repr__(self):
         return '<{} - {}>'.format(self.__class__.__name__, self.intent_type)
 
 
+def fuzzyfinder(name, entities):
+    """Semi fuzzy matching function."""
+    matches = []
+    pattern = '.*?'.join(name)
+    regex = re.compile(pattern)
+    for entity in entities:
+        match = regex.search(entity)
+        if match:
+            matches.append((len(match.group()), match.start(), entity))
+    return [x for _, _, x in sorted(matches)]
+
+
+class ServiceIntentHandler(IntentHandler):
+    """Intent handler registration."""
+
+    domain = None
```

Comment:
Would it make sense to take these in the constructor? That way you don't have to specify 3 classes in `components/__init__.py`.

Reply:
Do you mean by extending/creating a different async_register?

Reply:
```python
class ServiceIntentHandler(IntentHandler):

    def __init__(self, domain, service, response):
        self.domain = domain
        self.service = service
        self.response = response
```

Reply:
Well, when you put it that way :-) Thanks again.